### PR TITLE
fix: handle Claude Code 2.x JSON array response in summarization

### LINKED ISSUE #750: Summary generation fails: cannot unmarshal array from Claude Code 2.x JSON output
### What happened?

`entire explain --checkpoint <id> --generate` fails with a JSON parsing error. Entire expects a single JSON object from Claude CLI but Claude Code 2.x returns a JSON **array** when invoked with `--output-format json`.

```
$ entire explain --checkpoint 336b6c705d35 --generate
2026/03/22 00:32:21 INFO generating checkpoint summary
failed to generate summary: failed to generate summary: failed to parse claude CLI response: json: cannot unmarshal array into Go value of type summarize.claudeCLIResponse
```

The root cause: Claude Code 2.x `--print --output-format json` returns an array of objects (`[{type:"system",...}, {type:"assistant",...}, {type:"result",...}]`) instead of a single object. The Go struct `summarize.claudeCLIResponse` tries to unmarshal this array as a single object, which fails.

Auto-summarization on commit (`strategy_options.summarize.enabled: true` in `.entire/settings.json`) is also silently broken for the same reason — summaries are never generated.

### Steps to reproduce

1. Have Claude Code 2.x installed
2. Enable summarize in `.entire/settings.json`:
   ```json
   { "enabled": true, "strategy_options": { "summarize": { "enabled": true } } }
   ```
3. Make a commit (checkpoint is created without summary)
4. Run `entire explain --checkpoint <id> --generate`
5. Error: `failed to parse claude CLI response: json: cannot unmarshal array into Go value of type summarize.claudeCLIResponse`

You can verify the format change by running:
```bash
echo "hello" | claude --print --output-format json --max-turns 1
# Returns: [{...}, {...}, {...}]  ← array, not object
```

### Entire CLI version

Entire CLI 0.5.1 (d46fdc24), Go version: go1.26.1, OS/Arch: linux/amd64

### OS and architecture

Linux 6.6.87.2-microsoft-standard-WSL2 x86_64 (WSL2)

### Agent

Claude Code 2.1.81

### Terminal

Bash (WSL2)

### Logs / debug output

```bash
$ entire explain --checkpoint 336b6c705d35 --generate
2026/03/22 00:32:21 INFO generating checkpoint summary
failed to generate summary: failed to generate summary: failed to parse claude CLI response: json: cannot unmarshal array into Go value of type summarize.claudeCLIResponse
```

### Additional context

- Related to #334 (same `explain --generate` command, different error)
- The fix would likely involve parsing the JSON array and extracting the `result` object (last element with `type: "result"`) instead of trying to unmarshal the entire response as a single object
- Git version: 2.47.2, shell: bash

### PR DESCRIPTION
**Pull request by areporeporepo**
## Summary

Claude Code 2.x changed `--print --output-format json` to return a JSON array of messages (`[{type:"system",...}, {type:"result", result:"..."}]`) instead of a single object. This breaks `entire explain --generate` and auto-summarization on commit, since the Go code tries to unmarshal the array into a single `claudeCLIResponse` struct.

This PR updates `parseClaudeCLIResult()` to try the array format first (extracting the element with `type: "result"`), falling back to the single-object format for Claude Code 1.x backward compatibility.

Fixes #750

## What changed

- Added `Type` field to `claudeCLIResponse` struct
- Extracted parsing logic into `parseClaudeCLIResult()` that handles both formats
- Array format (2.x): iterates elements, returns the one with `type: "result"`
- Single-object format (1.x): fallback path, unchanged behavior

## Test plan

- [x] `mise run fmt` — no formatting issues
- [x] `mise run lint` — 0 issues
- [x] `mise run test` — all tests pass
- [x] New unit test: `TestParseClaudeCLIResult` — covers both formats, edge cases (no result element, invalid JSON, empty result)
- [x] New integration test: `TestClaudeGenerator_ArrayResponse` — full Generate() flow with 2.x array format
- [x] New error test: `TestClaudeGenerator_ArrayResponseNoResult` — array without result element
- [x] Existing tests pass unchanged (1.x format backward compat)

### CODE REVIEW

File: cmd/entire/cli/summarize/claude.go [hunk 1]
```diff
@@ -133,15 +136,14 @@ func (g *ClaudeGenerator) Generate(ctx context.Context, input Input) (*checkpoin
...
-	// Parse the CLI response
-	var cliResponse claudeCLIResponse
-	if err := json.Unmarshal(stdout.Bytes(), &cliResponse); err != nil {
-		return nil, fmt.Errorf("failed to parse claude CLI response: %w", err)
+	// Parse the CLI response.
+	// Claude Code 2.x returns a JSON array of messages; extract the "result" element.
+	// Claude Code 1.x returns a single JSON object; try that as a fallback.
+	resultJSON, err := parseClaudeCLIResult(stdout.Bytes())
+	if err != nil {
+		return nil, err
```
[the remark this answers is not in the document]
Comment (areporeporepo) [reply cmd/entire/cli/summarize/claude.go]: Fixed — Generate() now wraps with fmt.Errorf for consistent grep-able prefix.

File: cmd/entire/cli/summarize/claude.go [hunk 2]
```diff
@@ -154,6 +156,33 @@ func (g *ClaudeGenerator) Generate(ctx context.Context, input Input) (*checkpoin
...
+//   - Claude Code 1.x: single JSON object with a "result" field
+func parseClaudeCLIResult(data []byte) (string, error) {
+	// Try array format first (Claude Code 2.x)
+	var responses []claudeCLIResponse
+	if err := json.Unmarshal(data, &responses); err == nil {
+		for _, r := range responses {
+			if r.Type == "result" && r.Result != "" {
+				return r.Result, nil
+			}
+		}
```
[the remark this answers is not in the document]
Comment (areporeporepo) [reply cmd/entire/cli/summarize/claude.go:176]: Fixed — added hasEmptyResultElement tracking in the array path. Now returns 'contained empty result' (matching 1.x behavior) instead of the generic 'no result element' message.

File: cmd/entire/cli/summarize/claude.go [hunk 3]
```diff
@@ -67,7 +67,10 @@ type ClaudeGenerator struct {
 }
 
 // claudeCLIResponse represents the JSON response from the Claude CLI.
+// Claude Code 1.x returns a single object: {"result": "..."}
+// Claude Code 2.x returns an array: [{type:"system",...}, {type:"result", result:"..."}]
 type claudeCLIResponse struct {
+	Type   string `json:"type"`
 	Result string `json:"result"`
 }
```
[the remark this answers is not in the document]
Comment (areporeporepo) [reply cmd/entire/cli/summarize/claude.go:75]: Fixed — doc comment now uses valid JSON with quoted keys.

File: cmd/entire/cli/summarize/claude_test.go [hunk 4]
```diff
@@ -247,6 +308,70 @@ func TestClaudeGenerator_MarkdownCodeBlock(t *testing.T) {
...
+		{
+			name:        "invalid JSON",
+			input:       `not json at all`,
+			expectError: "parse claude CLI response",
+		},
+		{
+			name:        "single object with empty result",
+			input:       `{"result": ""}`,
+			expectError: "empty result",
+		},
```
[the remark this answers is not in the document]
Comment (areporeporepo) [reply cmd/entire/cli/summarize/claude_test.go:349]: Added — new test case "array with empty result element" covers this path and expects the "empty result" error.
